packages/astro/src/runtime/server/transition.ts

```diff
@@ -22,78 +21,80 @@ export function createTransitionScope(result: SSRResult, hash: string) {
+}
+
+type Entries<T extends Record<string, any>> = Iterable<[keyof T, T[keyof T]]>
+
+const getAnimations = (name: TransitionAnimationValue) => {
+	if (name === 'fade') return fade();
+	if (name === 'slide') return slide();
+	if (typeof name === 'object') return name;
+}
+
 export function renderTransition(
 	result: SSRResult,
 	hash: string,
 	animationName: TransitionAnimationValue | undefined,
 	transitionName: string
 ) {
-	let animations: TransitionDirectionalAnimations | null = null;
-	switch (animationName) {
-		case 'fade': {
-			animations = fade();
-			break;
-		}
-		case 'slide': {
-			animations = slide();
-			break;
-		}
-		default: {
-			if (typeof animationName === 'object') {
-				animations = animationName;
+	// Default to `fade` (similar to `initial`, but snappier)
+	if (!animationName) animationName = 'fade';
+	const scope = createTransitionScope(result, hash);
+	const name = transitionName ? toValidIdent(transitionName) : scope;
+	const sheet = new ViewTransitionStyleSheet(scope, name);
+	
+	const animations = getAnimations(animationName);
+	if (animations) {
+		for (const [direction, images] of Object.entries(animations) as Entries<typeof animations>) {
+			for (const [image, rules] of Object.entries(images) as Entries<typeof animations[typeof direction]>) {
+				sheet.addAnimationPair(direction, image, rules);
 			}
 		}
+	} else if (animationName === 'none') {
+		sheet.addAnimationRaw('old', 'animation: none; opacity: 0; mix-blend-mode: normal;')
+		sheet.addAnimationRaw('new', 'animation: none; mix-blend-mode: normal;')
 	}
 
-	const scope = createTransitionScope(result, hash);
+	result._metadata.extraHead.push(markHTMLString(`<style>${sheet.toString()}</style>`));
+	return scope;
+}
 
-	// Default transition name is the scope of the element, ie HASH-1
-	if (!transitionName) {
-		transitionName = scope;
-	}
+class ViewTransitionStyleSheet {
+	private modern: string[] = []
+	private fallback: string[] = []
 
-	const styles = markHTMLString(`<style>[data-astro-transition-scope="${scope}"] {
-	view-transition-name: ${transitionName};
-}
-	${
-		!animations
-			? ``
-			: // Regular animations
-			  `
-::view-transition-old(${transitionName}) {
-	${stringifyAnimation(animations.forwards.old)}
-}
-[data-astro-transition-fallback=old] [data-astro-transition-scope="${scope}"] {
-	${stringifyAnimation(animations.forwards.old)}
-}
+	constructor(private scope: string, private name: string) {}
 
-::view-transition-new(${transitionName}) {
-	${stringifyAnimation(animations.forwards.new)}
-}
-[data-astro-transition-fallback=new] [data-astro-transition-scope="${scope}"] {
-	${stringifyAnimation(animations.forwards.new)}
-}
+	toString() {
+		const { scope, name } = this;
+		const [modern, fallback] = [this.modern, this.fallback].map(rules => rules.join(''));
+		return [`[data-astro-transition-scope="${scope}"] { view-transition-name: ${name}; }`, this.layer(modern), fallback].join('')
```

Comment:
View Transition pseudos are now wrapped in `@layer astro` so that user CSS will always take precedence.